services/repository-managers/src/test/java/org/sagebionetworks/repo/web/controller/JSONEntityHttpMessageConverterTest.java

```diff
@@ -222,4 +233,189 @@ public void testPLFM_2079() throws Exception{
+		JSONObjectAdapter schema = new JSONObjectAdapterImpl();
+		// properties is a map of String to JsonSchema
+		schema.put("properties", properties);
+		JSONObjectAdapter adapter = new JSONObjectAdapterImpl();
+		adapter.put("schema", schema);
+		adapter.put("concreteType", "org.sagebionetworks.repo.model.schema.CreateSchemaRequest");
+		return adapter;
+	}
+	
+	@Test
+	public void testValidateJSONEntityValid() throws Exception {
+		// set up
+		JSONObjectAdapter adapter = setUpValidJSONEntity();
+		String beforeJsonString = adapter.toJSONString();
+		CreateSchemaRequest entity = new CreateSchemaRequest(adapter);
+		// test, no exception should be thrown
+		JSONEntityHttpMessageConverter.validateJSONEntity(entity, beforeJsonString);
+	}
+	
+	
+	@Test
+	public void testValidateJSONEntityNormal() throws Exception {
+		// set up
+		JSONObjectAdapter adapter = setUpInvalidJSONEntity();
+		String beforeJsonString = adapter.toJSONString();
+		CreateSchemaRequest entity = new CreateSchemaRequest(adapter);
+		// test
+		String message = assertThrows(IllegalArgumentException.class, () -> {
+			JSONEntityHttpMessageConverter.validateJSONEntity(entity, beforeJsonString);
+		}).getMessage();
+		assertEquals(message, "JSON Element in Entity is Unsupported: notPartOfSpecification");
+	}
+	
+	
+	@Test
+	public void testValidateJSONEntityEmbeddedSchema() throws Exception {
+		// set up
+		JSONObjectAdapter adapter = setUpInvalidJSONEntityEmbeddedSchema();
+		String beforeJsonString = adapter.toJSONString();
+		CreateSchemaRequest entity = new CreateSchemaRequest(adapter);
+		// test
+		String message = assertThrows(IllegalArgumentException.class, () -> {
+			JSONEntityHttpMessageConverter.validateJSONEntity(entity, beforeJsonString);
+		}).getMessage();
+		assertEquals(message, "JSON Element in Entity is Unsupported: notPartOfSpecification");
+	}
+	
+	@Test
+	public void testValidateJSONEntityArrayOfSchemas() throws Exception {
+		// set up
+		JSONObjectAdapter adapter = setUpInvalidJSONEntityArrayOfSchemas();
+		String beforeJsonString = adapter.toJSONString();
+		CreateSchemaRequest entity = new CreateSchemaRequest(adapter);
+		// test
+		String message = assertThrows(IllegalArgumentException.class, () -> {
+			JSONEntityHttpMessageConverter.validateJSONEntity(entity, beforeJsonString);
+		}).getMessage();
+		assertEquals(message, "JSON Element in Entity is Unsupported: notPartOfSpecification");
+	}
+	
+	@Test
+	public void testValidateJSONEntityMapOfSchemas() throws Exception {
+		// set up
+		JSONObjectAdapter adapter = setUpInvalidJSONEntityMapOfSchemas();
+		String beforeJsonString = adapter.toJSONString();
+		CreateSchemaRequest entity = new CreateSchemaRequest(adapter);
+		String message = assertThrows(IllegalArgumentException.class, () -> {
+			JSONEntityHttpMessageConverter.validateJSONEntity(entity, beforeJsonString);
+		}).getMessage();
+		assertEquals(message, "JSON Element in Entity is Unsupported: notPartOfSpecification");
+	}
+	
+	@Test

```

Comment:
read tests need two cases:
on the set and invalid
would be invalid if on the set but valid since it is not. (would throw if on the set)